AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Sync/SubscriptionSync/ReconcileAndLocalSave/ReconcileAndLocalSaveOperation.swift

```diff
@@ -252,21 +254,38 @@ class ReconcileAndLocalSaveOperation: AsynchronousOperation {
     private func saveMetadata(storageAdapter: StorageEngineAdapter,
                               inProcessModel: AppliedModel) {
         log.verbose(#function)
+
+        /// Do a local metadata query before saving to check if the `AppliedModel` is of `create` or
+        /// `update` MutationType in perspective of local store
+        let existsLocally: Bool
+        do {
+            let localMetadata = try storageAdapter.queryMutationSyncMetadata(for: remoteModel.model.id)
+            existsLocally = localMetadata == nil
```

Comment:
if localMetadata == nil, which means there is no localmetadata and then existsLocally is true ? this sounds it's backwards

Reply:
Updated as
```swift
existsLocally = localMetadata != nil
```
so as updated according to this [comment](https://github.com/aws-amplify/amplify-ios/pull/758#discussion_r502555866)